pyomo/contrib/doe/doe.py

```diff
@@ -1190,18 +1200,20 @@ def _fix_design(self, m, design_val, fix_opt=True, optimize_option=None):
             # Get Pyomo variable object
             cuid = pyo.ComponentUID(name)
             var = cuid.find_component_on(m)
-            if fix_opt:
-                # If fix_opt is True, fix the design variable
-                var.fix(design_val[name])
-            else:
-                # Otherwise check optimize_option
-                if optimize_option is None:
-                    # If optimize_option is None, unfix all design variables
-                    var.unfix()
+
+            if var.ctype is Var:
+                if fix_opt:
+                    var.fix(design_val[name])
                 else:
-                    # Otherwise, unfix only the design variables listed in optimize_option with value True
```

Comment:
Why did you remove the comments explaining the logic here?

Reply:
This is added back